tests/testthat/test-explanation.R

```diff
@@ -0,0 +1,105 @@
+library(testthat)
+library(shapr)
+library(xgboost)
+
+test_that("Test functions in explanation.R", {
+
+  # Load data -----------
+  data("Boston", package = "MASS")
+  x_var <- c("lstat", "rm", "dis", "indus")
+  y_var <- "medv"
+
+#  x_train <- as.matrix(tail(Boston[, x_var], 50))
+  y_train <- tail(Boston[, y_var], 50)
+  x_test <- as.matrix(head(Boston[, x_var], 2))
+
+  #setwd("./tests/testthat") # Uncomment when running manually
+
+  # Prepare the data for explanation. Path needs to be relative to testthat directory in the package
+  explainer <- readRDS(file = "test_objects/shapley_explainer_obj.rds")
+
+  # Creating list with lots of different explainer objects
+  Ex.list <- list()
+
+  # Using Random number generator settings as in R version 3.5.0. Seed set internally for reproducability
+  suppressWarnings(RNGversion("3.5.0")) # Testthat gives the warning for using "non-uniform 'Rounding' sampler". Suppress this to complete test
+
+  # Ex 1: Explain predictions (gaussian)
+  Ex.list[[1]] <- explain(x_test, explainer, approach = "gaussian", prediction_zero = mean(y_train))
+
+  # Ex 2: Explain predictions (copula)
+  Ex.list[[2]] <- explain(x_test, explainer, approach = "copula", prediction_zero = mean(y_train))
+
+  # Ex 3: Explain predictions (empirical, independence):
+  empirical_settings <- list(type = "independence", fixed_sigma_vec = 0.1, w_threshold = 0.95)
+  Ex.list[[3]] <- explain(x_test, explainer, approach = "empirical", prediction_zero = mean(y_train), type = "independence")
+
+  # Ex 4: Explain predictions (empirical, fixed sigma)
+  empirical_settings <- list(type = "fixed_sigma", fixed_sigma_vec = 0.1, w_threshold = 0.95)
+  Ex.list[[4]] <- explain(x_test, explainer, approach = "empirical", prediction_zero = mean(y_train), type = "fixed_sigma")
+
+
+  # Ex 5: Explain predictions (empirical, AICc)
+  empirical_settings <- list(
```

Comment:
Could you just remove everything that is `empirical_settings`. Only used in the old code.

Reply:
Haha, good point! Fixed in feb4f30.